### PR TITLE
Marketplace deal page

### PR DESCRIPTION
**Pull request by GrandSchtroumpf**
Create the deal page for marketplace
Hook creation of movie
Improve contract query

### CODE REVIEW

File: apps/catalog/catalog/src/app/marketplace/deal/negociation/negociation.component.ts [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+  payment$ = this.query.activeVersion$.pipe(map(displayPaymentSchedule));
+  
+  oldVersions$ = this.query.oldVersionsView$;
+  versionColumns = versionColumns;
+  initialVersionColumns = ['date', 'offer', 'status'];
+
+  constructor(private query: ContractQuery) { }
+
+  ngOnInit() {
+  }
```
Comment (phpgeekfr) [inline apps/catalog/catalog/src/app/marketplace/deal/negociation/negociation.component.ts]: same as above

File: apps/catalog/catalog/src/app/marketplace/deal/signed/signed.component.ts [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+  styleUrls: ['./signed.component.scss'],
+  changeDetection: ChangeDetectionStrategy.OnPush
+})
+export class SignedComponent implements OnInit {
+
+  constructor() { }
+
+  ngOnInit() {
+  }
+
```
Comment (phpgeekfr) [inline apps/catalog/catalog/src/app/marketplace/deal/signed/signed.component.ts]: same

File: apps/catalog/catalog/src/app/marketplace/deal/view/view.component.html [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+  <header fxLayout="row" fxLayoutAlign="space-between start" fxLayoutGap="24px" >
+    <div class="logo-container">
+      <img asset="archipel-content-icon.svg" type="logo" alt="Archipel Content logo" />
+    </div>
+    <div fxFlex fxLayout="column">
+      <span class="payment">{{ version.status | uppercase }}</span>
+      <h1>Achipel Content - {{ version.titleIds.length }} titles</h1>
+    </div>
+    <div fxLayout="column" fxLayoutAlign="start end" *ngIf="version.price as price" >
+      <span class="mat-caption"> {{ version.date | date }} </span>
```
Comment (phpgeekfr) [inline apps/catalog/catalog/src/app/marketplace/deal/view/view.component.html]: If you don't specify format (date:short for example), what is the default format?

File: libs/movie/src/lib/movie/+state/movie.model.ts [hunk 4]
```diff
@@ -65,18 +65,18 @@ export function createMovie(params: Partial<Movie> = {}): Movie {
     id: params.id,
     deliveryIds: [],
     _type: 'movies',
+    versionInfo: {languages:{}}, // TODO issue #1596
+    movieReview: [],
+    ...params,
```
Comment (phpgeekfr) [inline libs/movie/src/lib/movie/+state/movie.model.ts]: nice!

File: libs/movie/src/lib/movie/+state/movie.model.ts [hunk 5]
```diff
@@ -106,9 +106,11 @@ export function createMoviePromotionalElements(
 ): MoviePromotionalElements {
   return {
     trailer: [],
-    banner: createPromotionalElement(params.banner),
-    poster: [],
     still_photo: [],
+    ...params,
+    // We want a default poster as we look for the first one
+    poster: params.poster.length ? params.poster : [createPromotionalElement()],
```
Comment (phpgeekfr) [inline libs/movie/src/lib/movie/+state/movie.model.ts]: good

File: libs/movie/src/lib/movie/+state/movie.store.ts [hunk 6]
```diff
@@ -12,4 +12,7 @@ export class MovieStore extends EntityStore<MovieState> {
     super();
```
Comment (phpgeekfr) [inline libs/movie/src/lib/movie/+state/movie.store.ts]: if child constructor is empty, do you need to declare it ? Can't you just let it call the parent constructor ?

File: libs/movie/src/lib/movie/guards/movie-contract.guard.ts [hunk 7]
```diff
@@ -14,6 +32,17 @@ export class MovieContractGuard extends CollectionGuard<MovieState> {
...
+      // Reset the store everytime the movieId changes.
+      tap(_ => this.store.reset()),
+      switchMap(contract => {
+        // Filter movieIds before the query to relieve it.
+        const organizationMovieIds = this.organizationQuery.getActive().movieIds;
+        const movieIds = contract.titleIds.filter(titleId => organizationMovieIds.includes(titleId));
+
+        return awaitSyncQuery.call(this.service, movieListContractQuery(contract.id, movieIds))
+      }
+      )
```
Comment (phpgeekfr) [inline libs/movie/src/lib/movie/guards/movie-contract.guard.ts]: bad indent

File: libs/utils/src/lib/pipes/join.pipe.ts [hunk 8]
```diff
@@ -0,0 +1,8 @@
+import { Pipe, PipeTransform } from '@angular/core';
+
+@Pipe({ name: 'join', pure: true })
+export class JoinPipe implements PipeTransform {
+  transform(input:Array<any>, separator = ','):string {
+    return input.join(separator);
+}
+}
```
Comment (phpgeekfr) [inline libs/utils/src/lib/pipes/join.pipe.ts]: indent

File: apps/catalog/catalog/src/app/marketplace/deal/view/view.component.html (hunk 3 above)
Comment (GrandSchtroumpf) [reply apps/catalog/catalog/src/app/marketplace/deal/view/view.component.html]: 'mediumDate': equivalent to 'MMM d, y' (Jun 15, 2015).

File: libs/movie/src/lib/movie/+state/movie.store.ts (hunk 6 above)
Comment (GrandSchtroumpf) [reply libs/movie/src/lib/movie/+state/movie.store.ts]: Yes, it should be ok.

**Pull request merged**